Cargo.toml

```diff
@@ -33,14 +33,15 @@ async-trait = { version = "0.1" }
 clap = { version = "4.5" }
 env_logger = { version = "0.10.1" }
 futures = { version = "0.3.30" }
+lazy_static = { version = "1.5.0" }
 log = { version = "0.4.20" }
 json5 = { version =  "0.4.1" }
 serde = { version = "1.0.154", features = ["derive"] }
 serde_json = { version = "1.0.94" }
 uuid = { version = "1.7.0" }
-tokio = { version = "1.35.1", default-features = false }
+tokio = { version = "1.35.1", default-features = false, features = ["rt", "rt-multi-thread", "sync", "time"] }
```

Comment:
As a general pattern, I tend to declare at the workspace level with `default-features = false` and then enable features at the package level. That way everybody just gets what they need ...

Reply:
Yup, I agree that would be the way to go normally. I'll outline here why I didn't go that route.

* I want to allow us to compile the workspace project without relying on bringing in the vsomeip or Zenoh at all, if someone is interested in only working with / on the `up-streamer` generic library.
  * In particular... the vsomeip one. It's annoying at the moment to compile due to requiring us to supply the locations of the C++ headers of the machine. This is something I should look into fixing: opened issue here: https://github.com/eclipse-uprotocol/up-transport-vsomeip-rust/issues/33.
* A Cargo workspace does not allow us to pick and choose members to compile based on feature flags.
* A Cargo workspace does not allow us to optionally include dependencies based on feature flags.

This left me with the not great solution of pushing these dependencies down into each crate and use the same feature flags across the three crates: `example-streamer-uses`, `up-linux-streamer`, `up-linux-streamer-plugin`.

When the up-transport-vsomeip compilation story is less bad, I'd like to do as you suggest.